### ISSUE TITLE
consider using pillow instead of opencv for resizing

### ISSUE THREAD

**Issue by rom1504** [rom1504/img2dataset#8]
probably easier to install

but first bench it

**Comment by woctezuma** [comment]
> but first bench it

Maybe useful: https://github.com/uploadcare/pillow-simd#benchmarks

**Comment by rom1504** [comment]
Yeah
https://python-pillow.org/pillow-perf/ is saying that opencv is the same speed as pillow simd at resizing.
But no comparison with recent versions of pillow

**Comment by borisdayma** [comment]
There is a popular library for image augmentation in deep learning: [Albumentations](https://github.com/albumentations-team/albumentations).

They have interesting benchmarks on their website and leverage OpenCV.

What is also nice is that they consider also many edge cases:
* resize only when needed
* rounding errors that would change the size by a pixel
* handle more than 4 channels

Many methods used here are already implemented. For example [SmallestMaxSize](https://albumentations.ai/docs/api_reference/augmentations/geometric/resize/#albumentations.augmentations.geometric.resize.SmallestMaxSize) corresponds to `resize_keep_ratio`.

If you're interested, I can propose a PR.

**Comment by rom1504** [comment]
yeah I agree it looks pretty good.
Do you think that would also work for the "white border" resize method ?

**Comment by borisdayma** [comment]
Yes, with `PadIfNeeded`: https://albumentations.ai/docs/api_reference/augmentations/transforms/#albumentations.augmentations.transforms.PadIfNeeded

**Comment by rom1504** [comment]
cool!
I would gladly accept a PR for this then.
